taxi/tests/tests_search.py

```diff
@@ -0,0 +1,174 @@
+from django.test import TestCase
+from django.urls import reverse
+from django.contrib.auth import get_user_model
+from taxi.models import Manufacturer, Car
+
+User = get_user_model()
+
+
+class ManufacturerSearchViewTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user(
+            username="testuser",
+            password="testpass"
+        )
+
+        Manufacturer.objects.create(name="Toyota", country="Japan")
+        Manufacturer.objects.create(name="Ford", country="USA")
+        Manufacturer.objects.create(name="Tesla", country="USA")
+
+    def setUp(self):
+        self.client.login(
+            username="testuser",
+            password="testpass"
+        )
+
+    def test_manufacturer_search_results(self):
+        response = self.client.get(reverse(
+            "taxi:manufacturer-list"),
+            {"manufacturer": "Ford"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["manufacturer_list"]), 1)
+        self.assertEqual(
+            response.context["manufacturer_list"][0].name,
+            "Ford"
+        )
+
+    def test_manufacturer_search_no_results(self):
+        response = self.client.get(reverse(
+            "taxi:manufacturer-list"),
+            {"manufacturer": "Honda"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            len(response.context["manufacturer_list"]), 0
+        )
+
+    def test_manufacturer_search_partial_match(self):
+        response = self.client.get(
+            reverse("taxi:manufacturer-list"),
+            {"manufacturer": "T"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            len(response.context["manufacturer_list"]), 2
```

Comment:
Ensure that the manufacturer search view logic supports partial matches, as this test assumes partial matching functionality.